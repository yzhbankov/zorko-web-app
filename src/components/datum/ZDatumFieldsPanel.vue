<template>
  <v-card>
    <v-card-text>
      <v-subheader>Fields</v-subheader>
      <v-list>
        <v-list-tile
          avatar
          v-for="field in fields"
          :key="field.text">
          <v-list-tile-content>
            <v-list-tile-title>{{ field.text }}</v-list-tile-title>
          </v-list-tile-content>
        </v-list-tile>
      </v-list>
    </v-card-text>
  </v-card>
</template>

<script>
export default {
  name: 'ZDatumFieldsPanel',
  props: {
    fields: {
      type: Array
    }
  }
}
</script>
