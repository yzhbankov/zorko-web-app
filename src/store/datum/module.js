import state from './state'
import * as mutations from './mutations'
import * as actions from './actions'

export default ({namespaced}) => ({
  namespaced,
  state: state(),
  mutations,
  actions
})
