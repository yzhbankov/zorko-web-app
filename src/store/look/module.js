import state from './state'
import * as actions from './actions'
import * as mutations from './mutations'

export default ({namespaced}) => ({
  namespaced,
  state: state(),
  actions,
  mutations
})
