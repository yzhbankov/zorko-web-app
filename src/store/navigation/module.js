import state from './state'
import * as mutations from './mutations'

export default ({namespaced}) => ({
  namespaced,
  state: state(),
  mutations
})
