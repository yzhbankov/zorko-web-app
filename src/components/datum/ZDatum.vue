<template>
  <v-layout wrap>
    <v-flex xs3>
      <ZDatumFieldsPanel :fields="datum.fields"/>
    </v-flex>
    <v-flex xs8 style="margin-left: 10px">

    </v-flex>
  </v-layout>
</template>

<script>
import ZDatumFieldsPanel from './ZDatumFieldsPanel.vue'

import {createNamespacedHelpers} from 'vuex'
const {mapState} = createNamespacedHelpers('datum/')

export default {
  name: 'ZDatum',
  props: {
    name: {
      type: String
    }
  },
  components: {
    ZDatumFieldsPanel
  },
  computed: {
    ...mapState(['datum'])
  },
  created () {
    this.$store.dispatch({
      type: 'datum/loadDatum',
      name: this.name
    })
  }
}
</script>

