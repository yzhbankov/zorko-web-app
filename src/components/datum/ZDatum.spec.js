import {createLocalVue, shallow} from 'vue-test-utils'
import {createRenderer} from 'vue-server-renderer'
import Vuex from 'vuex'
import ZDatum from './ZDatum.vue'
import ZDatumFieldsPanel from './ZDatumFieldsPanel.vue'
import module from '../../store/datum/module'

const localVue = createLocalVue()
const datumModule = module({namespaced: true})

localVue.use(Vuex)

describe('ZDatum.vue', () => {
  let store
  let actions

  beforeEach(() => {
    actions = {
      loadDatum: jest.fn()
    }
    store = new Vuex.Store({
      state: {},
      modules: {
        datum: {
          ...datumModule,
          actions
        }
      }
    })
  })

  it('initialize', () => {
    const wrapper = shallow(ZDatum, {store, localVue})

    expect(wrapper.vm).toBeTruthy()
  })

  it('loads datum on create', () => {
    shallow(ZDatum, {store, localVue})

    expect(actions.loadDatum).toHaveBeenCalled()
  })

  it('has same shallow HTML structure', () => {
    const renderer = createRenderer()
    const wrapper = shallow(ZDatum, {
      store,
      localVue
    })

    renderer.renderToString(wrapper.vm, (err, str) => {
      if (err) throw new Error(err)
      expect(str).toMatchSnapshot()
    })
  })

  it('passes fields to ZDatumFieldsPanel', () => {
    const fields = [
      {
        'text': 'Name'
      },
      {
        'text': 'Miles_per_Gallon'
      },
      {
        'text': 'Cylinders'
      }
    ]
    const state = {
      ...datumModule.state,
      datum: {
        fields
      }
    }
    store = new Vuex.Store({
      state: {},
      modules: {
        datum: {
          ...datumModule,
          actions,
          state
        }
      }
    })
    const datumWrapper = shallow(ZDatum, {
      store,
      localVue
    })

    const datumFieldsPanelWrapper = datumWrapper.find(ZDatumFieldsPanel)

    expect(datumFieldsPanelWrapper.vm.$props.fields).toEqual(fields)
  })
})
