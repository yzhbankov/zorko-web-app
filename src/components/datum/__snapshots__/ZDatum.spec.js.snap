// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ZDatum.vue has same shallow HTML structure 1`] = `
<v-layout wrap="">
    <v-flex xs3="">
        <!---->
    </v-flex>
    <v-flex xs8="" style="margin-left:10px;"></v-flex>
</v-layout>
`;
